ckan/lib/plugins.py

```diff
@@ -524,3 +524,31 @@ def activity_template(self):
         return 'organization/activity_stream.html'
 
 _default_organization_plugin = DefaultOrganizationForm()
+
+
+class DefaultTranslation(object):
+    def directory(self):
+        '''Change the directory of the *.mo translation files
+
+        The default implementation assumes the plugin is
+        ckanext/myplugin/plugin.py and the translations are stored in
+        i18n/
+        '''
+        import os
+        return os.path.abspath(os.path.join(os.path.dirname(__file__),
```

Comment:
`return os.path.join(os.path.dirname(inspect.getfile(self.__class__)), '../../', 'i18n')`

should give me what I want, as self will still refer to the plugin class